### PR TITLE
Feat(Release): GET VULNERABILITIES from Release by API #1571

### PR DESCRIPTION
**Pull request by tuannn2**
Signed-off-by: Nguyen Nhu Tuan <tuan2.nguyennhu@toshiba.co.jp>

[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?

Issue: #1571

### How To Test?
- Get vulnerability information through Release API
- curl  'https://sw360.org/resource/api/releases/{releasesID}/vulnerabilities’​
- Output:
 ```
{
  "_embedded": {
    "sw360:vulnerabilities": [
      {
        "externalId": "CVE-2014-6726",
        "_links": {
          "self": {
            "href": "http://localhost:8080/resource/api/vulnerabilities/CVE-2014-6726"}
        }
      },
      {
        "externalId": "CVE-2012-5816",
        "_links": {
          "self": {
            "href": "http://localhost:8080/resource/api/vulnerabilities/CVE-2012-5816"}
        }
      }
    ]
  },
  "_links": {
    "curies": [
      {
        "href": "http://localhost:8080/resource/docs/{rel}.html",
        "name": "sw360",
        "templated": true
      }
    ]
  }
}
```
> How should these changes be tested by the reviewer?
> Have you implemented any additional tests?

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Comment by afsahsyeda** [comment]
Hi @tuannn2 ,

Please remove the extra line "===Links" from the Rest API Docs
image

File: rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ReleaseSpecTest.java [hunk 1]
```diff
@@ -323,8 +327,50 @@ public void before() throws TException, IOException {
...
+        vulDto1.setComment("Lorem Ipsum");
+        vulDto1.setExternalId("23105");
+        vulDto1.setProjectRelevance("APPLICABLE");
+        vulDto1.setIntReleaseId("3765276512");
+        vulDto1.setIntReleaseName("Angular 2.3.0");
+        vulDto1.setAction("Update to Fixed Version");
+        vulDto1.setPriority("1 - critical");
+        vulDtos.add(vulDto1);
+
+        VulnerabilityCheckStatus vulnCheckStatus = new VulnerabilityCheckStatus();
```
[the remark this answers is not in the document]
Comment (tuannn2) [reply rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ReleaseSpecTest.java]: I checked again and found it was not needed, I deleted them.
Thank you!

**Comment by tuannn2** [comment]
Thank you for your reviewing, @afsahsyeda 
I have fixed it according to your review and answer some of your questions.
Please check it!

File: rest/resource-server/src/docs/asciidoc/releases.adoc [hunk 2]
```diff
@@ -376,3 +376,19 @@ include::{snippets}/should_document_get_usedbyresource_for_release/curl-request.
 
 ===== Example response
 include::{snippets}/should_document_get_usedbyresource_for_release/http-response.adoc[]
+
+[[resources-release-get-release-vulnerabilities]]
+==== Listing release vulnerabilities
+
+A `GET` request will get vulnerabilities of a single release.
+
+include::{snippets}/should_document_get_release_vulnerabilities/links.adoc[]
```
Comment (afsahsyeda) [inline rest/resource-server/src/docs/asciidoc/releases.adoc:385]: Please add this line under line no. 394 with a "===== Links" heading, similar to the other sections in this doc

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java [hunk 3]
```diff
@@ -749,7 +757,7 @@ private void setDataCveReferences(Set<String> cveReferences, Vulnerability vulne
             vulnerability.setCveReferences(cveReferenceList);
         }
     }
-
+
```
Comment (afsahsyeda) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java:760]: Please remove the tab space on line no. 760

**Comment by afsahsyeda** [comment]
Hi @tuannn2 ,

I have added 2 more review comments. Please have a look

**Comment by tuannn2** [comment]
Thanks, @afsahsyeda
I have fixed it. Please check it!

**Pull request merged**